version: '3.9'
services:
  traefik:
    container_name: traefik
    image: traefik:2.9.6
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./config/traefik.yml:/traefik.yml:ro
      - ./config/acme_letsencrypt.json:/acme_letsencrypt.json
      - ./config/dynamic.conf.yml:/dynamic.conf.yml
      - /var/log/crowdsec/traefik/:/var/log/traefik/
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.traefik.entryPoints=http"
      - "traefik.http.routers.traefik.middlewares=traefik-auth@file,default@file"
      - "traefik.http.routers.traefik.service=api@internal"
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
    networks:
      - default
    hostname: traefik
    ports:
      - "80:80"
      - "443:443"

  pihole:
    container_name: pihole
    image: pihole/pihole:2023.01.3
    ports:
      - "53:53/tcp"
      - "53:53/udp"
      - "8085:80/tcp"
    environment:
      - TZ=Europe/Berlin
      - FTLCONF_LOCAL_IPV4=192.168.1.11
      - PIHOLE_DNS_=9.9.9.9;149.112.112.112
      - WEBPASSWORD=${WEBPASSWORD}
    volumes:
       - ./config/etc-pihole/:/etc/pihole/
       - ./config/etc-dnsmasq.d/:/etc/dnsmasq.d/
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.pihole.entryPoints=http"
      - "traefik.http.services.pihole.loadbalancer.server.port=80"
    networks:
      - default
    cap_add:
      - NET_ADMIN
    restart: unless-stopped

networks:
  default:
    name: proxy-net
